lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -625,6 +625,15 @@ data ApiPoolId
     | ApiPoolId PoolId
     deriving (Eq, Generic, Show)
 
+instance FromText AccountPublicKey where
+    fromText txt = case xpubFromText (T.encodeUtf8 txt) of
+        Left _ ->
+            Left $ TextDecodingError $ unwords $
+            [ "AccountPublicKey: unable to deserialize ShelleyKey from json. "
+            , "Expecting hex-encoded string of 128 characters."]
```

Comment:
The error message shouldn't mention:

- `ShelleyKey` (that's an internal data-type, not really suitable for public interfaces)
- `json` (it's actually not deserializing from JSON here, although we are probably using this instance also in JSON instances). 

I'd suggest something more aligned with other parser from hex-encoded bytestring:

_"Invalid account public key: expecting a hex-encoded value that is 64 bytes in length."_

Reply:
yes, my bad. I move it directly from FromJSON. corrected : https://github.com/input-output-hk/cardano-wallet/pull/1421/commits/ddec074ddadbddb70b0cfbd8899efa145b9c0dc7